src/twisted/internet/_sslverify.py

```diff
@@ -1591,10 +1591,14 @@ def __init__(self,
             self._options |= SSL.OP_NO_TICKET
         self.dhParameters = dhParameters
 
-        try:
-            self._ecCurve = _OpenSSLECCurve(_defaultCurveName)
-        except NotImplementedError:
-            self._ecCurve = None
+        self._ecCurve = None
+        # Only select a curve for OpenSSL < 1.0.2, in 1.0.2+ we use the
+        # automatic mode that supports multiple curves
+        if SSL.OPENSSL_VERSION_NUMBER < 0x10002000:
```

Comment:
I was surprised to see that this was green, since Twisted ends up installing `cryptography` 2.1.3, which should include OpenSSL 1.1.

It turns out we have [a Python 3.3 Travis environment with OpenSSL 1.0.1f](https://travis-ci.org/twisted/twisted/jobs/307037924#L84)  The `cryptography` project doesn't ship `manylinux1` wheels for Python 3.3, so this job builds `cryptography` 2.1.3 against OpenSSL 1.0.1 and runs the code in this block.

That's fine for this PR but bad in general because Python 3.3 has reached its end of life and Twisted will probably drop support for it.  We'll need builders with OpenSSL 1.0.1 if we're going to support it going forward.